Remove publish queueing feature

kt3k opened the pull request:
- This PR enables AutoReconnect option of paho client.
- This PR removes queueing feature of publishing messages because it's handled by paho (with `AutoReconnect: true`)
- This PR also removes auto re-subscription feature because subscription is persisted in mqtt broker of AWS IoT (see https://docs.aws.amazon.com/iot/latest/developerguide/mqtt-persistent-sessions.html)
  - We can reuse this persisted subscription if we set `CleanSession: true`.

ramaining tasks:
- [x] remove auto-reconnect
- [x] clean up subscription feature.
- [x] update unit tests

closes #38

at-wat commented:
https://github.com/seqsense/aws-iot-device-sdk-go/blob/a86ad5673524a4ca4dfe29c2a3f949fe80e8845c/v2/device.go#L103
should be true to enable paho.mqtt's reconnect feature.
If it is not true, queue looks cleared on new connection.

kt3k commented:
@at-wat Thank you for the important advises. I'll update the behavior around the reconnection.

kt3k commented:
@at-wat 
- Removed auto reconnection.
- Removed auto re-subscription.
- Kept subscription queueing.
- Added retry loop for first connection.
- Added option `CleanSession: false`
- Updated test cases.

kt3k commented:
@at-wat ping

at-wat commented:
I tested this with actual AWS IoT.

```json
# upper is latest
{
  "status": "dead"
}
# Stop device
...
{
  "status": "tick 16"
}
{
  "status": "tick 15"
}
{
  "status": "tick 11"
}
{
  "status": "tick 13"
}
{
  "status": "tick 14"
}
{
  "status": "tick 12"
}
{
  "status": "tick 10"
}
{
  "status": "tick 8"
}
{
  "status": "tick 9"
}
{
  "status": "dead"
}
# Disconnect device network at here and reconnect
{
  "status": "tick 5"
}
{
  "status": "tick 4"
}
{
  "status": "tick 3"
}
{
  "status": "tick 2"
}
{
  "status": "tick 1"
}
{
  "status": "tick 0"
}
```
Messages 6 and 7 which were sent between network down and MQTT client disconnect detection seems lost. (master branch looks correctly buffer and retransmit them.)

at-wat commented on v2/connection.go:
```diff
@@ -71,76 +52,33 @@ func connectionHandler(c *DeviceClient) {
-	token := s.cli.cli.Publish(d.Topic, s.cli.opt.Qos, s.cli.opt.Retain, d.Payload)
-	go func() {
-		token.Wait()
-		if token.Error() != nil {
```
maybe, this error handling is still required for https://github.com/seqsense/aws-iot-device-sdk-go/pull/39#issuecomment-521516827

kt3k commented:
@at-wat I created small programs to test the behaviors, but I couldn't find such message dropping behavior.

[Here](https://gist.github.com/kt3k/bcb00def16ea27ec3ffb46b67880203b) are the logs and programs I used to test the above. The order of the messages were shuffled but eventually the all messages seemed to come to the receiver.

BTW I used the setting like the below in go.mod:
```
replace github.com/seqsense/aws-iot-device-sdk-go/v2 v2.0.2 => /path/to/local/aws-iot-device-sdk-go/v2
```

at-wat commented on v2/options.go:
```diff
@@ -31,24 +31,18 @@ type ConnectionLostHandler func(*Options, *mqtt.ClientOptions, error)
-	ClientID                 string
-	Region                   string
-	BaseReconnectTime        time.Duration
-	MaximumReconnectTime     time.Duration
```
Seems better not to remove and pass the value to `paho.mqtt.ClientOptions`.
https://github.com/eclipse/paho.mqtt.golang/blob/93c8a06e1e6dfeb17dd18a44e3e2f4e1db778237/options.go#L69

default value in `paho.mqtt.ClientOptions` is quite large.
```go
		MaxReconnectInterval:    10 * time.Minute,
```
https://github.com/eclipse/paho.mqtt.golang/blob/93c8a06e1e6dfeb17dd18a44e3e2f4e1db778237/options.go#L108

kt3k commented:
@at-wat Addressed the above 2 points. PTAL.

at-wat commented:
@kt3k thanks. I'll quickly test it with AWS IoT.
